Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,10 +380,16 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        X = BasicStats(self.data[:, self.attr_x], self.attr_x)
+        Y = BasicStats(self.data[:, self.attr_y], self.attr_y)
```

Comment:
@wadhwasahil The `warning` call is fine where it is; it shows the warning depending on the attribute selection. But the _`BasicStats` call_ is the one that better be moved somewhere where it is computed only once per data, like @lanzagar said.